### ISSUE TITLE
deprecate `__coverage__` in favor of babel-plugin-istanbul

### ISSUE THREAD

**Issue by bcoe** [dtinth/babel-plugin-__coverage__#42]
Using @dtinth's amazing work as a basis, we've merged `babel-plugin-__coverage__` into the module:

https://www.npmjs.com/package/babel-plugin-istanbul

Would it make sense to point folks towards this plugin?

**Comment by dtinth** [comment]
Congrats on the release of `babel-plugin-istanbul`! 🎉
Very excited to try it.

Sure. First please let me test it with some code base. 😄

**Comment by dtinth** [comment]
I’ve tried `babel-plugin-istanbul` on the codebase at work (big one) and found it to fail to compile a function declaration. It fails at [this line](https://github.com/istanbuljs/istanbul-lib-instrument/blob/44b789b2923f85ea1ce8ee50495e77ed3566d34f/src/visitor.js#L178) indicating that, somehow, `n.id` is `undefined` (cannot read property loc of undefined). I failed to isolate this problem, so that may be caused by an outdated npm dependency or conflicting plugins)

I also noticed that it performs quite slower, but I may be wrong. I will perform a more thorough analysis later.

**Comment by Coobaha** [comment]
I confirm that `babel-plugin-istanbul` is very slow.
with 143 files covered
`__coverage__ ` ends in 25s
and `istanbul` ends in 148.6s

**Comment by bcoe** [comment]
@gotwarlost any thoughts about @dtinth's exception, I see that for `FunctionExpression` you have this safety:

```js
                if (n.id) {
                    dloc = n.id.loc;
                }
                break;
```

Do we need a similar safety for `FunctionDeclaration`?

With regards to performance, just off the top of my head, is there a chance that the new approach misses the babel-cache and that we're instrumenting files multiple times? nyc generates a cache of instrumented files by default, which helps performance, I would be curious to see [v8-profiler](https://www.npmjs.com/package/v8-profiler) output for a slow test run, so that we can isolate the performance issue.

**Comment by queicherius** [comment]
I use a `tests.webpack.js` as the entry point for testing my client side files. When switching to `babel-plugin-istanbul`, it did not ignore the `tests.webpack.js` (`babel-plugin-__coverage__` does ignore it correctly):

```json
    "test": {
      "presets": ["es2015"],
      "plugins": [["__coverage__", {"ignore": ["spec.js", "*.spec.js", "src/config/*", "tests.webpack.js"]}]]
    },
```

**Comment by bcoe** [comment]
@queicherius could you give this a spin and let me know if it works:

https://github.com/istanbuljs/nyc#excluding-files

tldr; include/exclude rules should fall under an nyc config flag.

**Comment by queicherius** [comment]
@bcoe Ah yes thanks, that works. Sorry for the confusion.

**Comment by kt3k** [comment]
I found this case. Maybe the same as the one mentioned by @dtinth.

```sh
istanbul-test $ cat test.js
export default function () {}
istanbul-test $ ./node_modules/.bin/babel --plugins istanbul < test.js
TypeError: unknown: Cannot read property 'loc' of null
    at VisitState.insertFunctionCounter (~/tmp/istanbul-test/node_modules/babel-plugin-istanbul/node_modules/istanbul-lib-instrument/dist/visitor.js:217:32)
    at VisitState.coverFunction (~/tmp/istanbul-test/node_modules/babel-plugin-istanbul/node_modules/istanbul-lib-instrument/dist/visitor.js:327:10)
    at ~/tmp/istanbul-test/node_modules/babel-plugin-istanbul/node_modules/istanbul-lib-instrument/dist/visitor.js:303:15
    at Array.forEach (native)
    at VisitState.wrappedEntry (~/tmp/istanbul-test/node_modules/babel-plugin-istanbul/node_modules/istanbul-lib-instrument/dist/visitor.js:302:15)
    at NodePath._call (~/tmp/istanbul-test/node_modules/babel-cli/node_modules/babel-core/node_modules/babel-traverse/lib/path/context.js:76:18)
    at NodePath.call (~/tmp/istanbul-test/node_modules/babel-cli/node_modules/babel-core/node_modules/babel-traverse/lib/path/context.js:48:17)
    at NodePath.visit (~/tmp/istanbul-test/node_modules/babel-cli/node_modules/babel-core/node_modules/babel-traverse/lib/path/context.js:106:12)
    at TraversalContext.visitQueue (~/tmp/istanbul-test/node_modules/babel-cli/node_modules/babel-core/node_modules/babel-traverse/lib/context.js:167:16)
    at TraversalContext.visitSingle (~/tmp/istanbul-test/node_modules/babel-cli/node_modules/babel-core/node_modules/babel-traverse/lib/context.js:118:19)
```

**Comment by gotwarlost** [comment]
Please try with `istanbul-lib-instrument@1.1.0-alpha.2`

**Comment by bcoe** [comment]
@kt3k @dtinth I've published `babel-plugin-istanbul@1.0.1` with @gotwarlost's fixes for these out of bounds issues, mind checking it out?

**Comment by kt3k** [comment]
@gotwarlost @bcoe 
Confirmed it's fixed at v1.0.1.
Thank you!

**Comment by dtinth** [comment]
Sorry this took so long. This project has been deprecated.
